src/response.php

```diff
@@ -141,9 +148,18 @@ public function send_response($content_type=null, $encode_options=null, $respons
 	
 	if ($this->http_status == self::STATUS_NO_CONTENT) {
 		return;
-	}
+	} 
 	
-	echo $response;
+	if ($content_type == self::CONTENT_TYPE_JSONP) {
+		if (empty($jsonp_callback)) {
+			$jsonp_callback = self::JSONP_CALLBACK_DEFAULT;
+		}
+		// jsonp response
+		echo $jsonp_callback.'('.$response.')';
+	} else { 
```

Comment:
Please keep the main flow unindented, by returning after echo'ing the jsonp response.